pkg/querier/queryrange/roundtrip.go

```diff
@@ -238,15 +238,30 @@ func transformRegexQuery(req *http.Request, expr syntax.LogSelectorExpr) (syntax
 	return expr, nil
 }
 
+func SmallestPositiveNonZeroIntPerTenant(tenantIDs []string, f func(string) int) int {
```

Comment:
We should have sich a method already. In any case why not move it to the other methods in the validation package?